src/core/Modal/Modal.tsx

```diff
@@ -0,0 +1,46 @@
+import { Modal as MuiModal, Button } from '@material-ui/core';
+import React, { useState } from 'react';
+import useStyles from './styles';
+
+interface ModalProps {
```

Comment:
The base mui modal can be extended, here....that will provide you with onClose by default